lib/ex_unit/lib/ex_unit/diff.ex

```diff
@@ -558,15 +558,15 @@ defmodule ExUnit.Diff do
   end
 
   defp move_right({y, list1, [elem2 | rest2], {edit1, edit2, env}}) do
-    {y, list1, rest2, {edit1, [{:ins, elem2} | edit2], env}}
+    {y, list1, rest2, {edit1, [{:ins, build_elem(elem2)} | edit2], env}}
   end
 
   defp move_right({y, list1, [], edits}) do
     {y, list1, [], edits}
   end
 
   defp move_down({y, [elem1 | rest1], list2, {edit1, edit2, env}}) do
-    {y + 1, rest1, list2, {[{:del, elem1} | edit1], edit2, env}}
+    {y + 1, rest1, list2, {[{:del, build_elem(elem1)} | edit1], edit2, env}}
```

Comment:
I am worried this change is happening too deep in the algorithm. We should probably deal with converting to struct/map when building the diff script?

Reply:
Yes - but in the `:eq` branch it's happening on the same level

Reply:
I can try to move it where it is extracted from the diff

Reply:
I'ts enough to convert it in the `escape` function, it is called in the `diff_improper` when the non matched elem is  inserted on the right (line 401) and `list_script_to_diff` when it's in `:del` on left side on line 621